### PR TITLE
quicp2p based networking

### PR DESCRIPTION
**Pull request by davidrusu**
Simple networking between multiple nodes, with a repl like environment for issuing commands and querying state.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,6 +17,9 @@ sha2 = "0.8.1"
 serde = "1.0.106"
 bincode = "1.2.1"
 hex = "0.4.2"
+cmdr = { path = "../cmdr/cmdr" }
```
Comment (dan-da) [inline Cargo.toml:20]: what is this cmdr?

**Review by dan-da — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (davidrusu) [reply Cargo.toml:20]: It's a little lib I found that gives you a nice repl like interface, they hadn't pushed up a release recently hence the path dependency, I've requested they publish to crates.io soon https://github.com/mendelt/cmdr/issues/42

File: src/qp2p_proc.rs [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        self.shared.lock().unwrap().apply(op);
+    }
+
+    fn read(&self) -> HashSet<u8> {
+        self.shared.lock().unwrap().read()
+    }
+}
+
+#[derive(Debug)]
+struct Proc {
```
Comment (davidrusu) [inline src/qp2p_proc.rs:72]: This ended up being unused, remove before merging

File: Cargo.toml (hunk 1 above)
Comment (dan-da) [reply Cargo.toml:20]: ok, we should be able to use a github reference in Cargo.toml meanwhile.  makes it easier to build.

Comment (davidrusu) [reply Cargo.toml]: done ✔️

**Comment by davidrusu** [comment]
@dan-da I've replaced the path deps with git references and peers now reciprocate adds to the qp2p peer set.

**Pull request merged**